src/metabase/upload.clj

```diff
@@ -271,7 +267,8 @@
   (let [normalized-header   (map normalize-column-name header)
         unique-header       (map keyword (mbql.u/uniquify-names normalized-header))
         column-count        (count normalized-header)
-        col-name+type-pairs (->> (column-types-from-rows settings column-count rows)
+        initial-types       (repeat column-count nil)
+        col-name+type-pairs (->> (column-types-from-rows settings initial-types rows)
```

Comment:
We need a threading style guide to align around 😄 

In this case I think the threading helps because it makes it easier to see that map will be taking two collections.

This does feel like a particularly subjective case that might class with rules of thumb I've thrown at you 😆